idaes/core/util/parameter_sweep.py

```diff
@@ -0,0 +1,812 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+IDAES Parameter Sweep API and sequential workflow runner.
+"""
+
+import sys
+import json
+
+from pandas import DataFrame
+
+from pyomo.core import Param, Var
+from pyomo.environ import check_optimal_termination
+from pyomo.common.config import ConfigDict, ConfigValue, document_kwargs_from_configdict
+
+import idaes.logger as idaeslog
+from idaes.core.surrogate.pysmo.sampling import SamplingMethods, UniformSampling
+from idaes.core.util.exceptions import ConfigurationError
+
+__author__ = "Andrew Lee"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+# TODO: Timeouts
+# TODO: Re-initialize option/callback
+
+
+class ParameterSweepSpecification(object):
+    """Defines a set of input variables/parameters and values to be used in
+    a parameter sweep study.
+
+    Users can define inputs to be sampled, the number of samples and the method to
+    use to generate the samples (chosen from Pysmo's sampling methods) and then
+    call the generate_samples method to generate the desired set of samples.
+    """
+
+    # TODO: Consider supporting sampling from data sets in the future
+    def __init__(self):
+        self._inputs = {}
+        self._sampling_method = None
+        self._samples = None
+        self._sample_size = None
+
+    def add_sampled_input(
+        self,
+        name: str,
+        pyomo_path: str,
```

Comment:
For WaterTAP at least they allow these to be different. One potential use case would be to give the parameters more compact or meaningful names instead of the full Pyomo path. We could make `name` optional however (defaulting to the Pyomo path).

Reply:
I see the use case, but think only *requiring* one name is a cleaner API